core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -854,6 +854,9 @@ public int performBatchRecovery(Iterable<Translog.Operation> operations) {
         if (state != IndexShardState.RECOVERING) {
             throw new IndexShardNotRecoveringException(shardId, state);
         }
+        // We set active because we are now writing operations to the engine; this way, if we go idle after some time and become inactive,
+        // we still give sync'd flush a chance to run:
```

Comment:
minor note - this is only called when recovering from a primary on a replica. When a replica becomes inactive we don't sync flush (we do that just on the primary). The change is still good, but I think we should change the comment.